dashboard/streamlit_app.py

```diff
@@ -1,73 +1,177 @@
-"""Streamlit dashboard application code"""
+"""
+Streamlit dashboard application code
+
+Module contains code for connecting to postgres database (RDS)
+and using that data to create charts for data analysis.
+"""
+import sys
 import os
 from os import environ
-import streamlit as st
 import pandas as pd
-from psycopg2 import connect
+import streamlit as st
+import matplotlib.pyplot as plt
+import seaborn as sns
+from psycopg2 import connect, Error
 from psycopg2.extensions import connection
 from dotenv import load_dotenv
 
+CSV_COLUMNS = [
+    "entry_id", "species", "temperature", "soil_moisture",
+    "last_watered", "recording_taken", "sunlight", "botanist_name", "cycle"
+]
+
+
+def get_db_connection():
+    """Establishes a connection with the PostgreSQL database."""
+    try:
+        conn = connect(
+            user=environ.get("DATABASE_USERNAME"),
+            password=environ.get("DATABASE_PASSWORD"),
+            host=environ.get("DATABASE_IP"),
+            port=environ.get("DATABASE_PORT"),
+            database=environ.get("DATABASE_NAME"),)
+        print("Database connection established successfully.")
+        return conn
+    except Error as err:
+        print("Error connecting to database: ", err)
+        sys.exit()
+
+
+def get_live_database(conn: connection) -> pd.DataFrame:
+    """Retrieve plant data from the database and return as a DataFrame."""
+
+    query = """
+    SELECT p.plant_entry_id AS entry_id,
+           s.s_name AS species,
+           p.temperature,
+           p.soil_moisture,
+           p.last_watered AS last_watered,
+           p.recording_taken AS recording_taken,
+           sun.s_description AS sunlight,
+           b.b_name AS botanist_name,
+           c.cycle_name AS cycle
+    FROM plant p
+    LEFT JOIN species s ON p.species_id = s.species_id
+    LEFT JOIN sunlight sun ON p.sunlight_id = sun.sunlight_id
+    LEFT JOIN botanist b ON p.botanist_id = b.botanist_id
+    LEFT JOIN cycle c ON p.cycle_id = c.cycle_id;
```

Comment:
did you figure out a different way to get a day's data or is this an older file?